### PR TITLE
[PM-6788][PM-7755] add babel/preset-env and browserslist

### PR DESCRIPTION
**Pull request by willmartian**
## 🎟️ Tracking

- https://bitwarden.atlassian.net/browse/PM-7755
- https://bitwarden.atlassian.net/browse/PM-6788

## 📔 Objective

Add `@babel/preset-env` and `browserslist` to our build pipeline to support a wider browser audience. 

This PR uses the suggested default config for `browserslist` of `> 0.5%, last 2 versions, Firefox ESR, not dead`, which at this time has [the following coverage](https://browsersl.ist/#q=defaults):

image

### Steps taken:

- added a babel config file that uses `@babel/preset-env`
- updated each webpack config to use the babel config via `babel-loader`
- added a `browserslist` config file (used by `@babel/preset-env`) that specifies the desired browser targets

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Protected functional changes with optionality (feature flags)
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation (Confluence, contributing docs) or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .browserslistrc [hunk 1]
```diff
@@ -0,0 +1 @@
+> 0.5%, last 2 versions, Firefox ESR, not dead
```
Comment (Hinton) [inline .browserslistrc:1]: We should consider changing to `major versions`, expands desktop Safari from 17.4 and 17.5 to also include 16. And include maintained node versions.

```suggestion
> 0.5%, last 2 major versions, Firefox ESR, not dead, maintained node versions
```

Comment (willmartian) [reply .browserslistrc:1]: Yeah, sounds reasonable to me!

Comment (willmartian) [reply .browserslistrc:1]: That updates our coverage to the following:

image

Comment (willmartian) [reply .browserslistrc]: Updated again to `> 0.5%, last 3 major versions, Firefox ESR, not dead` since that expands coverage to Safari 15. 

image

**Review by Hinton — APPROVED** [review]
This looks good, Does it make any difference if we put `browserlistrc` in the actual `apps` directories or is it fine in the project root?

We should also try finding someone whom can test on older Safari versions.

**Comment by willmartian** [comment]
> This looks good, Does it make any difference if we put `browserlistrc` in the actual `apps` directories or is it fine in the project root?

It does work in the root and I assume we want a consistent config everywhere. 

> We should also try finding someone whom can test on older Safari versions.

We've done some initial testing and have gotten conflicting results. We will continue testing this on `main` and the dev environment to see if we can ascertain why that is.

**Pull request merged**